core/src/scene/styleContext.cpp

```diff
@@ -82,6 +82,77 @@ StyleContext::~StyleContext() {
     duk_destroy_heap(m_ctx);
 }
 
+void StyleContext::parseSceneGlobals(const YAML::Node& node, const std::string& key, int seqIndex,
+                                     duk_idx_t dukObject) {
+
+    switch(node.Type()) {
+        case YAML::NodeType::Scalar:
+            if (key.size() == 0) {
+                duk_push_string(m_ctx, node.Scalar().c_str());
+                duk_put_prop_index(m_ctx, dukObject, seqIndex);
+            } else {
+                auto nodeValue = node.Scalar();
+                if (nodeValue.compare(0, 8, "function") == 0) {
+                    duk_push_string(m_ctx, key.c_str()); // push property key
+                    duk_push_string(m_ctx, nodeValue.c_str()); // push function string
+                    duk_push_string(m_ctx, "");
+                    if (duk_pcompile(m_ctx, DUK_COMPILE_FUNCTION) == 0) { // compile function
+                        duk_put_prop(m_ctx, -3); // put {key: function()}
+                    } else {
+                        LOGW("Compile failed: %s\n%s\n---",
+                             duk_safe_to_string(m_ctx, -1),
+                             nodeValue.c_str());
+                        duk_pop(m_ctx); //pop error
+                        duk_pop(m_ctx); //pop key
+                    }
```

Comment:
Thank you for the comments, I can totally grok this!

Here's a small dilemma: For strings that "look like" a JS function, we try to compile them and if that doesn't work we don't add a value for that key at all. But what if I'm a very crafty user and I want a global property that is the string "function"? Right now we would try to compile it, fail, and then not create that property at all. 

On the one hand, we could trust our users and infer that if the string doesn't compile as a function, it should just be a string property. 

On the other hand, it might be more useful to provide stronger feedback if something that "looks like" a function doesn't compile. 

cc @bcamper What does JS do in a case like this?